File: full_case_study_solns/_e_ballbeam/simulink/hw15/ballbeamParamHW15.m
% ballbeam parameter file
addpath ./.. % adds the parent directory to the path
ballbeam_param % general parameters

% parameter for dirty derivative
P.sigma = 0.05;

P_in = tf([P.ell/(P.m2*P.ell^2/3+P.m1*(P.ell/2)^2)],[1,0,0]);
P_out = tf(-P.g,[1,0,0]);

figure(2), clf
bode(P_in)
title('Ballbeam - Inner Loop')
grid on

figure(3), clf
bode(P_out)
title('Ballbeam - Outer Loop')
grid on

File: full_case_study_solns/_e_ballbeam/python_old/hw15/ballbeamParamHW15.py
# Ball on Beam Parameter File
import sys
sys.path.append('..')  # add parent directory
import ballbeamParam as P
import control as cnt
from control import TransferFunction as tf
import matplotlib.pyplot as plt


# Compute inner and outer open-loop transfer functions
P_in = tf([P.length/(P.m2*P.length**2/3.0+P.m1*(P.length/2.0)**2)],
          [1, 0, 0])
P_out = tf(-P.g, [1, 0, 0])

# Plot the closed loop and open loop bode plots for the inner loop
plt.figure(3), cnt.bode_plot(P_in, dB=True)
plt.figure(4), cnt.bode_plot(P_out, dB=True)

# Closes plot windows when the user presses a button.
plt.pause(0.0001)  # not sure why this is needed for both figures to display
print('Press key to close')
plt.waitforbuttonpress()
plt.close()
